fvm/src/syscalls/crypto.rs

```diff
@@ -47,6 +48,50 @@ pub fn verify_signature(
         .map(|v| if v { 0 } else { -1 })
 }
 
+pub fn verify_bls_aggregate(
+    context: Context<'_, impl Kernel>,
+    num_signers: u32,
+    sig_off: u32,
+    pub_keys_off: u32,
+    digests_off: u32,
+) -> Result<i32> {
+    let pub_keys_len = num_signers * BLS_PUB_LEN as u32;
+    let digests_len = num_signers * BLS_DIGEST_LEN as u32;
+
+    let sig: [u8; BLS_SIG_LEN] = context
+        .memory
+        .try_slice(sig_off, BLS_SIG_LEN as u32)?
+        .try_into()
+        .expect("bls signature bytes slice-to-array conversion should not fail");
+
+    let pub_keys: Vec<[u8; BLS_PUB_LEN]> = context
+        .memory
+        .try_slice(pub_keys_off, pub_keys_len)?
+        .chunks(BLS_PUB_LEN)
+        .map(|pub_key_bytes| {
+            pub_key_bytes
+                .try_into()
+                .expect("bls public key bytes slice-to-array conversion should not fail")
+        })
+        .collect();
+
+    let digests: Vec<[u8; BLS_DIGEST_LEN]> = context
+        .memory
+        .try_slice(digests_off, digests_len)?
+        .chunks(BLS_DIGEST_LEN)
+        .map(|digest_bytes| {
+            digest_bytes
+                .try_into()
+                .expect("bls digest bytes slice-to-array conversion should not fail")
+        })
+        .collect();
```

Comment:
We should do this without copying (the user isn't paying for that copy yet). Unfortunately, that means re-implementing https://doc.rust-lang.org/std/primitive.slice.html#method.as_chunks (unstable), but that's not the end of the world.